components/NavigationBar.tsx

```diff
@@ -0,0 +1,175 @@
+      />
+
+      {showMobileNavbar && (
+        <MobileNavigation
+          showMobileNavbar={showMobileNavbar}
+          closeImgSrc={closeImgSrc}
+          pathname={pathname}
+          toggleMobile={toggleMobile}
+        />
+      )}
+    </div>
+  );
+};
+
+export default Navbar;
+
+type MobileNavigationProps = {
+  toggleMobile: () => void;
+  showMobileNavbar: boolean;
+  closeImgSrc: string;
+  pathname: string;
+};
+
+type DeskTopNavbarProps = Omit<MobileNavigationProps, 'closeImgSrc'> & {
+  openImgSrc: string;
+};
+
+type NavigationRoutesProps = {
+  pathname: string;
+  showMobileNavbar: boolean;
+};
+
+const NavigationRoutes: React.FC<NavigationRoutesProps> = ({
+  pathname,
+  showMobileNavbar,
+}) => {
+  return (
+    <>
+      {routes.map((route) => (
+        <Link href={route.path} key={route.label}>
+          <p
+            className={`mx-3.5 ${
+              showMobileNavbar ? 'mt-6' : ''
+            }  flex h-full items-center text-[0.875rem] font-normal text-white-500 dark:text-white-800 ${
+              pathname === route.path &&
+              'font-semibold text-[#0252CD] dark:text-[#428DFF]'
+            }`}
+          >
+            {route.label}
+          </p>
+        </Link>
+      ))}
+    </>
+  );
+};
+
+const DesktopNavbar: React.FC<DeskTopNavbarProps> = ({
+  toggleMobile,
+  showMobileNavbar,
+  openImgSrc,
+  pathname,
+}) => {
+  return (
+    <div className="flex items-center justify-around">
+      <div className="hidden items-center md:flex">
+        <NavigationRoutes
+          pathname={pathname}
+          showMobileNavbar={showMobileNavbar}
+        />
+      </div>
+      <div className="hidden md:flex">
+        <DownloadResume />
+        <div className="mx-[2.2rem] hidden h-[1.5rem] w-[1px] bg-white-500 dark:bg-white-500 md:flex" />
+        <ThemeSwitcher />
+      </div>
+
+      {!showMobileNavbar && (
+        <Image
+          src={openImgSrc}
+          width={30}
+          height={30}
+          alt="logo"
+          className="mr-24 flex md:hidden"
+          onClick={toggleMobile}
+        />
+      )}
+    </div>
+  );
+};
+
+const MobileNavigation: React.FC<MobileNavigationProps> = ({
+  toggleMobile,
+  showMobileNavbar,
+  closeImgSrc,
+  pathname,
+}) => {
+  return (
+    <div className="fixed right-0 z-30  flex h-full w-2/4 items-center dark:bg-black-200 md:hidden">
+      <Image
+        src={closeImgSrc}
+        width={30}
+        height={30}
+        alt="logo"
+        className="absolute right-9 top-9 cursor-pointer md:hidden"
+        onClick={toggleMobile}
+      />
+      <div className="mb-20 flex w-full flex-col p-8">
+        <NavigationRoutes
+          pathname={pathname}
+          showMobileNavbar={showMobileNavbar}
+        />
+        <div className="ml-2.5 mt-6">
+          <DownloadResume />
+        </div>
+        <div className="ml-3 mt-6 flex justify-start">
+          <ThemeSwitcher />
+        </div>
+      </div>
+    </div>
+  );
+};
```

Comment:
This shouldn't be in the same file, extract these into separate component files and import them here.

Reply:
Resolved

Reply:
Refactored.